### DISCUSSION TITLE
Reliable timestamp for when a scheduled job should have ran

### DISCUSSION THREAD

**Discussion by ryankauk** [taskforcesh/bullmq#676] [Q&A]
Hey I was wondering if there's anyway to get the timestamp of a scheduled job for when it should have ran in case of some type of failure. I would like to use it for determining a time range. 

**Assumptions:**
Lets say you scheduled a job to run every minute.
For simplicity sake lets say it started at 12:00pm and first one successfully ran.
You shutdown the worker and wait 2 minutes.
Start the worker back up. 

Is there a way to get the timestamp 12:01 pm for the next time the worker starts?

Seems job.timestamp would be offset to the future due to the mechanics of how scheduled jobs work based off delayed jobs.

**Comment by manast** [comment]
Not sure I understand exactly what you are looking after, but during the time workers are offline no accumulation of repeatable jobs is performed, so it will execute the 12:01 as soon as it goes online (with the onProcessed timestamp set to the moment it actually starts processing it) and then continue with the next job one minute after that one.

**Reply by ryankauk** [reply]
Yes, I'm looking for the timestamp the scheduled job was supposed to have run instead of when it was processed. If it's set to a cron to run every minute and the server shutdown for 5 minutes and ran at 12:05. I would like to get the 12:01 timestamp instead of 12:05. Is this something that is doable?
